lib/kubernetes-deploy/kubernetes_resource/service.rb

```diff
@@ -7,9 +7,11 @@ def sync(cache)
       super
       if exists? && selector.present?
         @related_deployments = cache.get_all(Deployment.kind, selector)
+        @related_statefulsets = cache.get_all(StatefulSet.kind, selector)
```

Comment:
I think this is fine for now, but if we find ourselves having to accommodate some other kind of workload we should consider aggregating things into a `@related_workloads = {}` to simplify processing

Reply:
Yep, tackling that in https://github.com/Shopify/kubernetes-deploy/pull/541/files#diff-bfab9900d04f0a61190aa04f896aed1fR10

Reply:
:+1: Yep just saw that and was on my way to delete this comment